cirq-core/cirq/interop/quirk/cells/input_rotation_cells_test.py

```diff
@@ -89,9 +89,9 @@ def test_input_rotation_cells():
     assert_url_to_circuit_returns(
         '{"cols":[["•","X^(-A/2^n)","inputA2"]]}',
         diagram="""
-0: ───@────────────
+0: ───@^(-A/2^2)───
```

Comment:
Why did the exponent move from the target to the ancilla? That's not where it should be.

Reply:
This is because the gate changed from being a `ControlledOperation(sub_type='X')` to a `CX` gate. The latter always prints the exponent on the large qubit, and not necessarily on the target. Eg:

```python
In [1]: import cirq
   ...: q = cirq.LineQubit.range(2)
   ...: gate = cirq.CX ** 0.25
   ...: c1 = cirq.Circuit(gate(q[0], q[1]))
   ...: c2 = cirq.Circuit(gate(q[1], q[0]))
   ...: print(c1, c2, sep = '\n---------\n')
0: ───@────────
      │
1: ───X^0.25───
---------
0: ───X────────
      │
1: ───@^0.25───
```

Reply:
Sent another PR to fix the behavior of `CX` and `CCX` gates - https://github.com/quantumlib/Cirq/pull/4462

Reply:
I don't want to break then unbreak the diagrams. Merge the relevant changes from the other PR into this one. Make zero breaking changes instead of two breaking changes that undo each other.

Reply:
Once the other PR is merged and this PR is rebased, this particular breaking change would go away.

Also, the two PRs / changes would independently always be breaking changes. We can't merge them into one PR and not have a breaking change. This is because:
1) The other PR breaks all diagrams using CX and CCX gates. This includes diagrams inside `cirq-core/cirq/ops/common_gates_test.py` which have nothing to do with this PR. 
2) This PR breaks diagrams where a `@ ---- S` would get plotted as `@ --- @ ** 0.5`. These breaking changes are also unavoidable and have nothing to do with the other PR. 

The other PR is approved and I'll merge once this discussion is resolved.

Reply:
I think the diagrammatic representations of gates are important and it's good that we have tests to ensure that we don't break them, but I don't think we should consider the diagrams to be part of the _stable_ interface for which we provide backwards compatibility guarantees. This strikes me as something like making guarantees about error messages so that users can match errors with regexes and we won't break them. That seems like an anti-pattern; users should match on error _types_ instead. And similarly cirq users should IMO check types and values in their tests, not string diagrams (unless they are testing their own diagram code for their own gates). I realize this means us, since we have a lot of internal tests that match circuit diagrams, but I think we should change that or be willing to update tests when upstream diagrams change.

Reply:
@Strilanc Ping. LMK your thoughts so we can merge the other pending PR and unblock this one as well.

Reply:
@Strilanc Ping. Gentle reminder.

Reply:
I submitted #4462  but there are more bugs related to drawing diagrams for controlled operations in other classes. See #4515